bgpd/bgp_zebra.c

```diff
@@ -356,6 +359,35 @@ static int bgp_interface_address_delete(ZAPI_CALLBACK_ARGS)
 		bgp_connected_delete(bgp, ifc);
 	}
 
+	addr = ifc->address;
+
+	if (bgp) {
+		/*
+		 * When we are using the v6 global as part of the peering
+		 * nexthops and we are removing it, then we need to
+		 * clear the peer data saved for that nexthop and
+		 * cause a re-announcement of the route.  Since
+		 * we do not want the peering to bounce.
+		 */
+		for (ALL_LIST_ELEMENTS(bgp->peer, node, nnode, peer)) {
+			afi_t afi;
+			safi_t safi;
+
+			if (addr->family == AF_INET)
+				continue;
+
+			if (!IN6_IS_ADDR_LINKLOCAL(&addr->u.prefix6)
+			    && memcmp(&peer->nexthop.v6_global,
+				      &addr->u.prefix6, 16)
+				       == 0) {
+				memset(&peer->nexthop.v6_global, 0, 16);
+				FOREACH_AFI_SAFI (afi, safi)
```

Comment:
this loop is same as bgp_announce_route_all().

Reply:
bgp_announce_route() introduced additional `force` attribute. In this case, it has to be passed to bgp_announce_route_all(). IMO, it's fine here with this change.